vispy/scene/canvas.py

```diff
@@ -241,73 +262,205 @@ def draw_visual(self, visual, event=None, viewport=None):
+                    # events that are not handled get passed to parent
+                    picked = picked.parent
+                    scene_event.visual = picked
+                else:
+                    picked = None
+            
         # If something in the scene handled the scene_event, then we mark
         # the original event accordingly.
         event.handled = scene_event.handled
 
+    def visual_at(self, pos):
+        """Return the visual at a given position
+
+        Parameters
+        ----------
+        pos : tuple
+            The position in logical coordinates to query.
+
+        Returns
+        -------
+        visual : instance of Visual | None
+            The visual at the position, if it exists.
+        """
+        tr = self.transforms.get_transform('canvas', 'framebuffer')
+        fbpos = tr.map(pos)[:2]
+
+        try:
+            id_ = self._render_picking(region=(fbpos[0]-10, fbpos[1]-10,
+                                               20, 20))
+            vis = VisualNode._visual_ids.get(id_[10, 10], None)
+        except RuntimeError:
+            # Don't have read_pixels() support for IPython. Fall back to
+            # bounds checking.
+            return self._visual_bounds_at(pos)
+        return vis
+
+    def _visual_bounds_at(self, pos, node=None):
+        """Find a visual whose bounding rect encompasses *pos*.
+        """
+        if node is None:
+            node = self.scene
+            
+        for ch in node.children:
+            hit = self._visual_bounds_at(pos, ch)
+            if hit is not None:
+                return hit
+        
+        if (not isinstance(node, VisualNode) or not node.visible or 
+                not node.interactive):
+            return None
+        
+        bounds = [node.bounds(axis=i) for i in range(2)]
+        
+        if None in bounds:
+            return None
+        
+        tr = self.scene.node_transform(node).inverse
+        corners = np.array([
+            [bounds[0][0], bounds[1][0]],
+            [bounds[0][0], bounds[1][1]],
+            [bounds[0][1], bounds[1][0]],
+            [bounds[0][1], bounds[1][1]]])
+        bounds = tr.map(corners)
+        xhit = bounds[:, 0].min() < pos[0] < bounds[:, 0].max()
+        yhit = bounds[:, 1].min() < pos[1] < bounds[:, 1].max()
+        if xhit and yhit:
+            return node
+
+    def visuals_at(self, pos, radius=10):
+        """Return a list of visuals within *radius* pixels of *pos*.
+        
+        Visuals are sorted by their proximity to *pos*.
+        
+        Parameters
+        ----------
+        pos : tuple
+            (x, y) position at which to find visuals.
+        radius : int
+            Distance away from *pos* to search for visuals.
+        """
+        tr = self.transforms.get_transform('canvas', 'framebuffer')
+        pos = tr.map(pos)[:2]
+
+        id = self._render_picking(region=(pos[0]-radius, pos[1]-radius,
```

Comment:
+1 for opening an issue that picking will fail for IPython. It would also be a good place do discuss the idea of having multiple mouse-interaction modes.